cmd/k8s-operator/sts.go

```diff
@@ -223,7 +223,7 @@ func statefulSetNameBase(parent string) string {
 			return base
 		}
 		base = base[:len(base)-1-excess]   // cut off the excess chars
-		if !strings.HasSuffix(base, "-") { // dash may have been cut by the generator
+		if !strings.HasSuffix(base, "-") { // re-instante the dash
```

Comment:
dash was definitely cut off in the previous step, no matter what the generator did